code/modules/vehicles/motorized_wheelchair.dm

```diff
@@ -0,0 +1,168 @@
+	var/speed = 2
+	var/power_efficiency = 1
+	var/power_usage = 100
+	var/panel_open = FALSE
+	var/list/required_parts = list(/obj/item/stock_parts/manipulator, 
+							/obj/item/stock_parts/manipulator,
+							/obj/item/stock_parts/capacitor)
+	var/obj/item/stock_parts/cell/power_cell
+
+/obj/vehicle/ridden/wheelchair/motorized/CheckParts(list/parts_list)
+	..()
+	refresh_parts()
+
+/obj/vehicle/ridden/wheelchair/motorized/proc/refresh_parts()
+	speed = 1 // Should never be under 1
+	for(var/obj/item/stock_parts/manipulator/M in contents)
+		speed += M.rating
+	for(var/obj/item/stock_parts/capacitor/C in contents)
+		power_efficiency = C.rating
+	var/datum/component/riding/D = GetComponent(/datum/component/riding)
+	D.vehicle_move_delay = round(CONFIG_GET(number/movedelay/run_delay) * 6.7) / speed
+
+/obj/vehicle/ridden/wheelchair/motorized/obj_destruction(damage_flag)
+	var/turf/T = get_turf(src)
+	for(var/atom/movable/A in contents)
+		A.forceMove(T)
+		if(isliving(A))
+			var/mob/living/L = A
+			L.update_mobility()
+	..()
+
+/obj/vehicle/ridden/wheelchair/motorized/driver_move(mob/living/user, direction)
+	if(istype(user))
+		if(!canmove)
+			return FALSE
+		if(!power_cell)
+			to_chat(user, "<span class='warning'>There seems to be no cell installed in [src].</span>")
+			canmove = FALSE
+			addtimer(VARSET_CALLBACK(src, canmove, TRUE), 20)
+			return FALSE
+		if(power_cell.charge < power_usage / max(power_efficiency, 1))			
+			to_chat(user, "<span class='warning'>The display on [src] blinks 'Out of Power'.</span>")
+			canmove = FALSE
+			addtimer(VARSET_CALLBACK(src, canmove, TRUE), 20)
+			return FALSE
+		if(user.get_num_arms() < arms_required)
+			to_chat(user, "<span class='warning'>You don't have enough arms to operate the motor controller!</span>")
+			canmove = FALSE
+			addtimer(VARSET_CALLBACK(src, canmove, TRUE), 20)
+			return FALSE
+		power_cell.use(power_usage / max(power_efficiency, 1))
+	return ..()
+
+/obj/vehicle/ridden/wheelchair/motorized/set_move_delay(mob/living/user)
+	return
+
+/obj/vehicle/ridden/wheelchair/motorized/post_buckle_mob(mob/living/user)
+	. = ..()
+	density = TRUE
+
+/obj/vehicle/ridden/wheelchair/motorized/post_unbuckle_mob()
+	. = ..()
+	density = FALSE
+
+/obj/vehicle/ridden/wheelchair/motorized/attack_hand(mob/living/user)
+	if(power_cell && panel_open)
+		power_cell.update_icon()
+		user.put_in_hands(power_cell)
+		power_cell = null
+		to_chat(user, "<span class='notice'>You remove the power cell from [src].</span>")
+		return
+	return ..()
+	
+/obj/vehicle/ridden/wheelchair/motorized/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		I.play_tool_sound(src)
+		panel_open = !panel_open
+		user.visible_message("<span class='notice'>[user] [panel_open ? "opens" : "closes"] the maintenance panel on [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the maintenance panel.</span>")
+		return
+	if(panel_open)
+		if(istype(I, /obj/item/stock_parts))
```

Comment:
is this check needed

Reply:
Below the power cell stuff it uses stock_parts, but I can move it.

Reply:
moved it to be after power cell stuff.